README.md

```diff
@@ -63,6 +63,10 @@ jobs:
 Require additional modules through composer for matrix entries. You do not need to quote the string for multiple requirements
 `composer_require_extra: silverstripe/widgets:^2 silverstripe/comments:^3`
 
+##### Composer install
+Use `composer install` rather than default `composer update`. Intended only for testing branches of `silverstripe/installer` or `silverstripe/recipe-kitchen-sink` where multiple pull-requests need to be tested together. If `true` will cause `composer_require_extra` input to be ignored. If `dynamic_matrix` input also is `true` then jobs will only be created for the PHP version defined in `config.platform.php` in composer.json (recommended), or the lowest supported PHP version. There will also be no `--prefer-lowest` jobs created. Default is false, enable with:
```

Comment:
Because you cannot use `composer require` if you want to respect an existing composer.lock file

> The result in matrix seems to be that it's just the simple matrix with a fixed php version, which doesn't seem consistent with what I had expected.

That's a requirement when using a composer.lock file - it's an explicitly fixed build

We cannot have reliably have multiple PHP versions because there is no guarantee that whatever dependencies that were installed are compatible with both PHP 7 and PHP 8

This is intended only for validating multiple pull-requests. My recommendation when doing it is to explicitly set `config.platform.php` in composer.json to guarantee that you get something installable in CI, otherwise you'll get dependencies that match whatever version of PHP you're running on your laptop, most likely PHP 8.1, which will match CMS 5 in CI (8.1), but not CMS 4 (7.4)

If you want multiple PHP versions, I think vendor-code-patchers needs to be used.  Of course that has its own limitations in that it cannot handle any changes in composer.json files in pull-requests.

On the balance of this, I think testing in a single version of PHP is fine because we generally get code failures in one version of PHP and not another.  Generally the type of failures we get between different version of PHP relate to being unable to install dependencies.

Reply:
To be clear, I have no problem with having a fixed php version - I understand that's necessary due to different versions of dependencies having different PHP support.

I guess ultimately anything that would require `composer_require_extra` can be done by just ensuring that dependency is in the composer.lock file, and if you only wanted it for a single job maybe you have to do multiple branches... so I guess this is okay. Will resolve conversation.